### PR TITLE
chore(snuba) - Matching snuba's internal query limit

### PR DESCRIPTION
**Pull request by wmak**
- This matches our limits with what snuba has internally
- Also moving this to a constant so if snuba changes the limit, we can
  match it easily

### CODE REVIEW

**Comment by dashed** [comment]
We should let the SNS team sign off on this.

**Comment by lynnagara** [comment]
This is a product change on the stats and discover endpoints. Do we know who is relying on the larger limits here.

**Comment by lynnagara** [comment]
The PR description is also misleading since the snuba limit is 10,000 not 1000

**Comment by wmak** [comment]
@lynnagara Hrm maybe I misunderstood something then. @fpacifici had mentioned to me here: https://www.notion.so/sentry/Q1-Backend-Plan-be735036516542b7a8f271f5073fc973?d=09eed214-e94d-4478-be9e-be3ee79bd20a#258a001a779548e4846097d63e8b3309

That the limit was 1,000.

@bowencai8 If this is done correctly (ie. matching the limit properly which I might not be right now :( ) there shouldn't be any impact to the user.

**Comment by lynnagara** [comment]
I don't have access to view that doc. Default is 1000, max is 10000 https://github.com/getsentry/snuba/blob/5e5de9e85b68fb931c7e9711db5c873774363dcf/snuba/query/schema.py#L104

**Comment by dashed** [comment]
What @lynnagara has said seems correct to me.  @fpacifici can you comment?

**Comment by fpacifici** [comment]
@lynnagara  is right on the number, I was wrong sorry.

Though I don't think it is a good idea to hardcode the maximum snuba limit and use it everywhere, expecially where you do not need it. If youdon't have a specific max size you need, please let Snuba pick the default (which is 1000), for the cases where you specifically need more (like stats) then feel free to use it.
But please avoid defaulting on the maximum when you do not need it, that would jsut be wasting resources.

File: src/sentry/tagstore/snuba/backend.py [hunk 1]
```diff
@@ -741,7 +742,7 @@ def get_tag_value_paginator_for_projects(
             conditions=conditions,
             orderby=order_by,
             # TODO: This means they can't actually paginate all TagValues.
-            limit=1000,
```
Comment (fpacifici) [inline src/sentry/tagstore/snuba/backend.py]: If this query worked with 1000, please leave it to 1000. if it worked, it means there is no reason for potentially load 10x more data.

**Review by fpacifici — CHANGES_REQUESTED** [review]

**Comment by wmak** [comment]
Perfect @fpacifici going to close this PR then! Thanks for the info :+1: I wanted to hardcode it for the queries where we were trying to get the maximum (eg. with timeseries)

**Comment by fpacifici** [comment]
The max you want for a time series depends on the query, not on the fact it is a time series.

**Comment by dashed** [comment]
@fpacifici would this be easy to determine?

**Comment by fpacifici** [comment]
Generally it should. That's the number of points you expect to render if it is a time series (you can calculate that from the bucket and start-end timestamp), but that depends on your use case. 
Still I would not overly complicate the solution, it is better using the default if that works for you even if it may load slightly more data than you need, then calculating it wrong. 

My concern was about taking the max and basically making it the new default.

**Comment by dashed** [comment]
Gotcha. 👍  Thanks for explaining.

**Pull request closed without merging**